docker/README.md

```diff
@@ -0,0 +1,128 @@
+# Dockerizing
+
+This video is a good resource for tips on how to optimize docker setup
+
+https://youtu.be/kG2vxYn547E?t=255 
+
+TLDR;
+
+[See slides here](https://speakerd.s3.amazonaws.com/presentations/55d8979b402d43f4a5a99cce56273579/containers.pdf)
+
+## Tip #1: Read and understand your base image
+
+There is an official [ruby image](https://hub.docker.com/_/ruby) on Docker Hub, but it comes with batteries included. 
+
+[Alpine](https://hub.docker.com/_/alpine) is a very slim linux that can be extended with ruby.
+
+## Tip #2: Combine install commands with cleanup
+
+👍👍👍
+```Docker
+RUN apt-get update -y \
+ && apt-get install -y -q mypackage \
+ && apt-get clean \
+ && rm -f /var/lib/apt/lists/*_* 
+```
+
+👎🤨😔
+```Docker
+RUN apt-get update -y
+RUN apt-get install -y -q mypackage
+RUN apt-get clean \
+ && rm -f /var/lib/apt/lists/*_* 
+```
+
+## Tip #3: Use a separate build stage
+
+Multi-stage docker files
+
+```Docker
+FROM my-base AS builder
+COPY . /app
+RUN apt-get install build-essential \
+ && bundle install --deployment
+
+FROM my-base
+COPY --from=builder /app /app
+```
+
+<img src="multi-stage-docker-files.png" alt="Multi-stage docker files" height="100">
+
+## Tip #4: Set the system locale
+
+```Docker
+RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen \
```

Comment:
Probably not the locale we want. Why is this useful?

Reply:
I copied it from somewhere else, without thinking about it. Actually, at that point I only thought the docker image would be used in development. I don't think it is useful.

Reply:
In the rails conf video, he says it is very important and often overlooked. Apparently you can in uncommon situations end up with ascii strings in Ruby, instead of utf8 strings (more so in earlier versions of Ruby/Rails than later). Unfortunately, he does not give an example of it.